### ISSUE TITLE
Add breadcrumbs to docs

### ISSUE THREAD

**Issue by Josh-Cena** [facebook/docusaurus#6096]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

Docs should have breadcrumbs in addition to sidebar.

### Has this been requested on Canny?

No

### Motivation

I've just built this feature for someone: https://github.com/PatelN123/Digital-Support-Notes/pull/1

It wasn't quite ergonomic, and I think it would make sense to make itself into the core.

### API design

It would be opt-in, something like `showBreadCrumbs` in the docs plugin. I think we can also set a metadata to help with SEO?

### Have you tried building it?

Yes, see that PR for how I did it.

### Self-service

- [X] I'd be willing to contribute this feature to Docusaurus myself.

**Comment by slorber** [comment]
Yes, we could add that option, and probably turn it on by default if it looks good. I'm fine with an opinionated design as long as it can be turned off

SEO metas should probably always be there even if the option is turned off?

`useBreadcrumb` in theme-common 👍 the lighter the UI component the better

There's already a component in Infima:

https://infima.dev/docs/components/breadcrumbs

Seems only used on the search page 😅 : https://docusaurus.io/search?q=doc

![image](https://user-images.githubusercontent.com/749374/146205552-29ab9409-a0d3-47ca-ad11-948a20097a26.png)

**Comment by christopherklint97** [comment]
Is it ok if I start working on this? Or will the implementation just be @Josh-Cena `useBreadcrumbs`?

**Comment by Josh-Cena** [comment]
@christopherklint97 You can start with `useBreadcrumbs` + a metadata in each Doc item, and then we can discuss if the UI feature is useful

**Comment by slorber** [comment]
> + a metadata in each Doc item

It is probably unnecessary to provide additional metadata here.

We now have a theme-common `useDocsSidebar()` (in `docsUtils.ts`), it should be possible to build on top of it a `useDocsBreadcrumb(docId)`

**Comment by Josh-Cena** [comment]
When I say metadata I don't mean props. I mean hooks + `<head><meta />`
